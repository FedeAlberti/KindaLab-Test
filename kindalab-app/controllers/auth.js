const {response} = require('express')
const bcryptjs = require('bcryptjs')
const User = require('../models/User')

let resp;

const newUser = async(req, res = response) =>{
    resp = res
    const{ email, password} = req.body
    console.log(req.body)
    try {
        let usero = await User.findOne({email});
        if(usero){
            return error400('mail already registered')
        }
        
        usero = new User(req.body)
        
        const salt = bcryptjs.genSaltSync();
        usero.password = bcryptjs.hashSync(password,salt);
        
        await usero.save();

        return success201(usero);

    } catch (error) {      
        return error500(error);
    }
}

const loginUser = async(req, res = response) =>{
    resp = res
    const{ email, password } = req.body
    try {

        const usero = await User.findOne({email});
        
        if(!usero){
            return error400('invalid user')
        }
        
        const validPassword = bcryptjs.compareSync(password, usero.password);

        if (!validPassword){
            return error400('invalid password')
        }

        return success201(usero)
        
    } catch (error) {
        error500(error);
    }

}

const success201 = (usero) =>{
        resp.status(201).json({
        ok:true,
        msg: 'Registered',
        uid:usero.id,
        name:usero.name
    })
}

const error400 = (msg) =>{
    resp.status(400).json({
        ok:false,
        msg: msg,
    })
}

const error500 = (error) =>{
    resp.status(500).json({
        ok:false,
        msg: 'Something goes bad',
        error:error
    })
}


module.exports = {
    newUser,
    loginUser
}