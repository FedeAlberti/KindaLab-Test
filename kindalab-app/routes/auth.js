const { Router } = require('express');
const { check } =require('express-validator')
const { validateFields } = require('../middlewares/validate-fields')
const { newUser, loginUser } = require('../controllers/auth');
const router =Router();

/*
    Route:
    host + /api/auth
*/

router.post(
    '/new',
     [
        check('name','Name is required ').not().isEmpty(),
        check('email','email is required ').isEmail(),
        check('password','The minimum password length is 6 characters').isLength({min:6}),
        validateFields,
     ] ,
      newUser);

router.post(
    '/',
    [
        check('email','email is required ').isEmail(),
        check('password','The minimum password length is 6 characters').isLength({min:6}),
        validateFields,
    ],
     loginUser);

module.exports = router;