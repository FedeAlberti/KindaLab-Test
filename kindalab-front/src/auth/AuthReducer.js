import { typesAuth } from '../types/typesAuth';


export const AuthReducer = (state={}, action) =>{
    debugger
    switch (action.type) {
        case typesAuth.login:
            return {
                ...action.payload,
                logged:true,
            }

        case typesAuth.logout:
            return{
                logged:false
            }
    
        default:
            return state;
            
    }
}