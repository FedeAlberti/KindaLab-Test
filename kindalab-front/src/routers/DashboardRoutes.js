import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'

import { FoodScreen } from '../components/FoodScreen'
import { Navbar } from '../components/UI/Navbar'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />
        <div className="container-fluid">
            <Switch>
                <Route exact path="/food" component={FoodScreen} />

                <Redirect to="/food" />
            </Switch>

        </div>
            
        </>
    )
}
