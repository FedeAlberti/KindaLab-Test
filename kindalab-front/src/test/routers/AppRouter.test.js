import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Test on AppRouter', () => {
    
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }


    test('Should show Login', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        
    });


    test('Should show FoodComponent', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Fede'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
        

    })
    
    

})
