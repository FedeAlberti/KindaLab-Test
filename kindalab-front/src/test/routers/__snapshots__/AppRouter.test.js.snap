// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AppRouter Should show Login 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lEmail"
              onChange={[Function]}
              placeholder="Correo"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lPassword"
              onChange={[Function]}
              placeholder="Contraseña"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rName"
              onChange={[Function]}
              placeholder="Nombre"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rEmail"
              onChange={[Function]}
              placeholder="Correo"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword1"
              onChange={[Function]}
              placeholder="Contraseña"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword2"
              onChange={[Function]}
              placeholder="Repita la contraseña"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
